core/src/main/scala/kafka/consumer/TopicCount.scala

```diff
@@ -104,6 +104,9 @@ private[kafka] object TopicCount extends Logging {
   def constructTopicCount(consumerIdString: String, filter: TopicFilter, numStreams: Int, zkUtils: ZkUtils, excludeInternalTopics: Boolean) =
     new WildcardTopicCount(zkUtils, consumerIdString, filter, numStreams, excludeInternalTopics)
 
+  override def equals(o: Any): Boolean = {
+    throw new KafkaException("can't use equals here")
+  }
```

Comment:
@jozi-k, we don't intend to merge this change. We are just checking if `equals` is ever invoked during the tests.